### ISSUE TITLE
Already disposed: URL=''

### ISSUE THREAD

**Issue by JPry** [hsz/idea-gitignore#366]
When IntelliJ was running an indexing operation, this error popped up. Here are the details from the dialog:

```
Already disposed: URL=''
java.lang.Throwable
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:132)
	at com.intellij.openapi.vfs.impl.VirtualFilePointerImpl.a(VirtualFilePointerImpl.java:94)
	at com.intellij.openapi.vfs.impl.VirtualFilePointerImpl.getFile(VirtualFilePointerImpl.java:71)
	at com.intellij.openapi.roots.impl.ContentEntryImpl.getFile(ContentEntryImpl.java:98)
	at com.intellij.openapi.roots.impl.RootModelBase.getContentRoots(RootModelBase.java:41)
	at com.intellij.openapi.roots.impl.ModuleRootManagerImpl.getContentRoots(ModuleRootManagerImpl.java:270)
	at com.intellij.openapi.roots.impl.FileIndexBase.getModuleContentAndSourceRoots(FileIndexBase.java:69)
	at com.intellij.openapi.roots.impl.ProjectFileIndexImpl.b(ProjectFileIndexImpl.java:69)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:48)
	at com.intellij.openapi.roots.impl.ProjectFileIndexImpl.a(ProjectFileIndexImpl.java:65)
	at com.intellij.openapi.roots.impl.ProjectFileIndexImpl.iterateContent(ProjectFileIndexImpl.java:54)
	at mobi.hsz.idea.gitignore.FilesIndexCacheProjectComponent.getFilesForPattern(FilesIndexCacheProjectComponent.java:159)
	at mobi.hsz.idea.gitignore.reference.IgnoreReferenceSet$IgnoreReference.innerResolveInContext(IgnoreReferenceSet.java:231)
	at com.intellij.psi.impl.source.resolve.reference.impl.providers.FileReference.innerResolve(FileReference.java:145)
	at com.intellij.psi.impl.source.resolve.reference.impl.providers.FileReference$MyResolver.resolve(FileReference.java:575)
	at com.intellij.psi.impl.source.resolve.reference.impl.providers.FileReference$MyResolver.resolve(FileReference.java:569)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:185)
	at com.intellij.psi.impl.source.resolve.reference.impl.providers.FileReference.multiResolve(FileReference.java:133)
	at mobi.hsz.idea.gitignore.codeInspection.IgnoreUnusedEntryInspection$1.visitEntry(IgnoreUnusedEntryInspection.java:79)
	at mobi.hsz.idea.gitignore.psi.IgnoreVisitor.visitEntryFile(IgnoreVisitor.java:19)
	at mobi.hsz.idea.gitignore.psi.impl.IgnoreEntryFileImpl.accept(IgnoreEntryFileImpl.java:20)
	at com.intellij.codeInspection.InspectionEngine.acceptElements(InspectionEngine.java:81)
	at com.intellij.codeInspection.InspectionEngine.createVisitorAndAcceptElements(InspectionEngine.java:69)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.a(LocalInspectionsPass.java:282)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.a(LocalInspectionsPass.java:250)
	at com.intellij.concurrency.ApplierCompleter.c(ApplierCompleter.java:133)
	at com.intellij.concurrency.ApplierCompleter.d(ApplierCompleter.java:222)
	at com.intellij.concurrency.ApplierCompleter.c(ApplierCompleter.java:151)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1147)
	at com.intellij.concurrency.ApplierCompleter.b(ApplierCompleter.java:105)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:556)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:501)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:66)
	at com.intellij.concurrency.ApplierCompleter.a(ApplierCompleter.java:116)
	at com.intellij.concurrency.ApplierCompleter.b(ApplierCompleter.java:96)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:142)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:242)
	at com.intellij.concurrency.ApplierCompleter.compute(ApplierCompleter.java:96)
	at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.pollAndExecCC(ForkJoinPool.java:1190)
	at java.util.concurrent.ForkJoinPool.helpComplete(ForkJoinPool.java:1879)
	at java.util.concurrent.ForkJoinPool.awaitJoin(ForkJoinPool.java:2045)
	at java.util.concurrent.ForkJoinTask.doJoin(ForkJoinTask.java:390)
	at java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:719)
	at java.util.concurrent.ForkJoinPool.invoke(ForkJoinPool.java:2616)
	at com.intellij.concurrency.JobLauncherImpl.invokeConcurrentlyUnderProgress(JobLauncherImpl.java:63)
	at com.intellij.concurrency.JobLauncher.invokeConcurrentlyUnderProgress(JobLauncher.java:57)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.a(LocalInspectionsPass.java:253)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.a(LocalInspectionsPass.java:223)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.collectInformationWithProgress(LocalInspectionsPass.java:132)
	at com.intellij.codeInsight.daemon.impl.ProgressableTextEditorHighlightingPass.doCollectInformation(ProgressableTextEditorHighlightingPass.java:83)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:70)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.b(PassExecutorService.java:438)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1153)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.d(PassExecutorService.java:431)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:556)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:501)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:66)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.c(PassExecutorService.java:430)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.a(PassExecutorService.java:406)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:142)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:242)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:404)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:165)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)


java.lang.Throwable
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:132)
	at com.intellij.openapi.vfs.impl.VirtualFilePointerImpl.a(VirtualFilePointerImpl.java:94)
	at com.intellij.openapi.vfs.impl.VirtualFilePointerImpl.getFile(VirtualFilePointerImpl.java:71)
	at com.intellij.openapi.roots.impl.ContentEntryImpl.getFile(ContentEntryImpl.java:98)
	at com.intellij.openapi.roots.impl.RootModelBase.getContentRoots(RootModelBase.java:41)
	at com.intellij.openapi.roots.impl.ModuleRootManagerImpl.getContentRoots(ModuleRootManagerImpl.java:270)
	at com.intellij.openapi.roots.impl.FileIndexBase.getModuleContentAndSourceRoots(FileIndexBase.java:69)
	at com.intellij.openapi.roots.impl.ProjectFileIndexImpl.b(ProjectFileIndexImpl.java:69)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:48)
	at com.intellij.openapi.roots.impl.ProjectFileIndexImpl.a(ProjectFileIndexImpl.java:65)
	at com.intellij.openapi.roots.impl.ProjectFileIndexImpl.iterateContent(ProjectFileIndexImpl.java:54)
	at mobi.hsz.idea.gitignore.FilesIndexCacheProjectComponent.getFilesForPattern(FilesIndexCacheProjectComponent.java:159)
	at mobi.hsz.idea.gitignore.reference.IgnoreReferenceSet$IgnoreReference.innerResolveInContext(IgnoreReferenceSet.java:231)
	at com.intellij.psi.impl.source.resolve.reference.impl.providers.FileReference.innerResolve(FileReference.java:145)
	at com.intellij.psi.impl.source.resolve.reference.impl.providers.FileReference$MyResolver.resolve(FileReference.java:575)
	at com.intellij.psi.impl.source.resolve.reference.impl.providers.FileReference$MyResolver.resolve(FileReference.java:569)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:185)
	at com.intellij.psi.impl.source.resolve.reference.impl.providers.FileReference.multiResolve(FileReference.java:133)
	at mobi.hsz.idea.gitignore.codeInspection.IgnoreUnusedEntryInspection$1.visitEntry(IgnoreUnusedEntryInspection.java:79)
	at mobi.hsz.idea.gitignore.psi.IgnoreVisitor.visitEntryFile(IgnoreVisitor.java:19)
	at mobi.hsz.idea.gitignore.psi.impl.IgnoreEntryFileImpl.accept(IgnoreEntryFileImpl.java:20)
	at com.intellij.codeInspection.InspectionEngine.acceptElements(InspectionEngine.java:81)
	at com.intellij.codeInspection.InspectionEngine.createVisitorAndAcceptElements(InspectionEngine.java:69)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.a(LocalInspectionsPass.java:282)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.a(LocalInspectionsPass.java:250)
	at com.intellij.concurrency.ApplierCompleter.c(ApplierCompleter.java:133)
	at com.intellij.concurrency.ApplierCompleter.d(ApplierCompleter.java:222)
	at com.intellij.concurrency.ApplierCompleter.c(ApplierCompleter.java:151)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1147)
	at com.intellij.concurrency.ApplierCompleter.b(ApplierCompleter.java:105)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:556)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:501)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:66)
	at com.intellij.concurrency.ApplierCompleter.a(ApplierCompleter.java:116)
	at com.intellij.concurrency.ApplierCompleter.b(ApplierCompleter.java:96)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:142)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:242)
	at com.intellij.concurrency.ApplierCompleter.compute(ApplierCompleter.java:96)
	at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.pollAndExecCC(ForkJoinPool.java:1190)
	at java.util.concurrent.ForkJoinPool.helpComplete(ForkJoinPool.java:1879)
	at java.util.concurrent.ForkJoinPool.awaitJoin(ForkJoinPool.java:2045)
	at java.util.concurrent.ForkJoinTask.doJoin(ForkJoinTask.java:390)
	at java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:719)
	at java.util.concurrent.ForkJoinPool.invoke(ForkJoinPool.java:2616)
	at com.intellij.concurrency.JobLauncherImpl.invokeConcurrentlyUnderProgress(JobLauncherImpl.java:63)
	at com.intellij.concurrency.JobLauncher.invokeConcurrentlyUnderProgress(JobLauncher.java:57)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.a(LocalInspectionsPass.java:253)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.a(LocalInspectionsPass.java:223)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.collectInformationWithProgress(LocalInspectionsPass.java:132)
	at com.intellij.codeInsight.daemon.impl.ProgressableTextEditorHighlightingPass.doCollectInformation(ProgressableTextEditorHighlightingPass.java:83)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:70)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.b(PassExecutorService.java:438)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1153)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.d(PassExecutorService.java:431)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:556)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:501)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:66)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.c(PassExecutorService.java:430)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.a(PassExecutorService.java:406)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:142)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:242)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:404)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:165)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)

```

**Comment by hsz** [comment]
@JPry Thank you for this report.

Are you able to reproduce this error every time? What IDE/OS do you use?

**Comment by Dragonrun1** [comment]
Happened to me when I was switching projects and told it to use the current window.

**Comment by JPry** [comment]
@hsz Unfortunately I have not seen this error repeat itself again, but I will see if there's any way to trigger it reliably.

**Comment by hsz** [comment]
Can I ask you to check if this issue is still present in v2.1.1?

**Comment by JPry** [comment]
@hsz Unfortunately I haven't been able to replicate the error.

I did install the v2.1.1 update. If I see any errors I'll be sure to report them.

**Comment by sergei-ivanov** [comment]
I suddenly started experiencing this issue, and the error kept popping up.

Plugin version 3.0.0.182
IntelliJ IDEA 2018.2.3 (Ultimate Edition)
Build #IU-182.4323.46, built on September 3, 2018
JRE: 1.8.0_152-release-1248-b8 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.15.0-33-generic

Looking at the stack trace, I decided to examine my project structure. And indeed there seemed to be a rogue empty content root in one of the modules. I think it appeared as a result of some other bug in the IDE, because that project suddenly disappeared at some point and I needed to reimport it from maven POM.

Deleting the rogue content root, refreshing Maven projects and restarting IDEA seems to have got rid of the problem.
